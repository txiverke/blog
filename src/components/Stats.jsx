// @flow

import React from 'react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'

import Loader from '../components/Loader'
import { normalizeVal } from '../utils/helpers'
import { loadStats } from '../actions/statsActionCreators'

const sections = [{ url: 'about-me' }, { url: 'posts' }, { url: 'projects'}]

class Stats extends React.PureComponent {
  props: {
    dispatch: Function,
    statistic: Statistic,
    authenticate: Auth,
    DIC: Object
  }

  componentDidMount() {
    const { dispatch, statistic, authenticate } = this.props

    if (Object.keys(statistic.data).length === 0 || authenticate.data) {
      dispatch(loadStats())
    }
  }


  getValue(section: string) {
    const { data } = this.props.statistic
    const users = data.users ? data.users : '0'
    const posts = data.posts ? data.posts : '0'
    const projects = data.projects ? data.projects : '0'

    switch (section) {
      case 'about-me': return normalizeVal(users)
      case 'posts': return normalizeVal(posts)
      case 'projects': return normalizeVal(projects)
      default: return '00'
    }
  }

  render() {
    const { authenticate, statistic, DIC } = this.props
    const url = authenticate.data ? '/admin': ''
    const label = authenticate.data ? DIC.EDIT : ''

    if (statistic.completed) {
      return (
        <div className="app-content">
        {sections.map(item => {
          const title = DIC[item.url.toUpperCase()]
          
          return (
            <article key={item.url} className="app-content-landing">
              <h2>
                <Link to={`${url}/${item.url}`}>
                  <span>{this.getValue(item.url)}</span>
                  <span className="bold">{`${label} ${title}`}</span>
                </Link>
              </h2>
            </article>
          )
        })}
        </div> 
      )
    }
    
    return <Loader msg={this.props.statistic.message} />
  }
}

const mapStateToProps = (state: Object) => ({ 
  statistic: state.statistic,
  authenticate: state.authenticate 
})

export const Unwrapped = Stats
export default connect(mapStateToProps)(Stats)