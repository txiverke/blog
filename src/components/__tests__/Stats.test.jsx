// @flow

import React from 'react'
import { shallow } from 'enzyme'
import { Unwrapped as UnwrappedStats } from '../Stats'

const properties = {
  dispatch: jest.fn(),
  statistic: {
    completed: true,
    data: {},
    message: 'test message',
    error: false
  },
  authenticate: {
    completed: true,
    data: 'token',
    message: 'test message',
    error: false
  },
  DIC: {}
}

describe('<Stats />', () => {
  it('should render itself', () => {
    const component = shallow(<UnwrappedStats {...properties} />)
    expect(component).toMatchSnapshot()
  })
})