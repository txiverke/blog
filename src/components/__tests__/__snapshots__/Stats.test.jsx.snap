// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Stats /> should render itself 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Stats
    DIC={Object {}}
    authenticate={
        Object {
            "completed": true,
            "data": "token",
            "error": false,
            "message": "test message",
          }
    }
    dispatch={[Function]}
    statistic={
        Object {
            "completed": true,
            "data": Object {},
            "error": false,
            "message": "test message",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <article
          className="app-content-landing"
>
          <h2>
                    <Link
                              replace={false}
                              to="/admin/about-me"
                    >
                              <span>
                                        00
                              </span>
                              <span
                                        className="bold"
                              >
                                        undefined undefined
                              </span>
                    </Link>
          </h2>
</article>,
        <article
          className="app-content-landing"
>
          <h2>
                    <Link
                              replace={false}
                              to="/admin/posts"
                    >
                              <span>
                                        00
                              </span>
                              <span
                                        className="bold"
                              >
                                        undefined undefined
                              </span>
                    </Link>
          </h2>
</article>,
        <article
          className="app-content-landing"
>
          <h2>
                    <Link
                              replace={false}
                              to="/admin/projects"
                    >
                              <span>
                                        00
                              </span>
                              <span
                                        className="bold"
                              >
                                        undefined undefined
                              </span>
                    </Link>
          </h2>
</article>,
      ],
      "className": "app-content",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "about-me",
        "nodeType": "host",
        "props": Object {
          "children": <h2>
            <Link
                        replace={false}
                        to="/admin/about-me"
            >
                        <span>
                                    00
                        </span>
                        <span
                                    className="bold"
                        >
                                    undefined undefined
                        </span>
            </Link>
</h2>,
          "className": "app-content-landing",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/admin/about-me"
>
              <span>
                            00
              </span>
              <span
                            className="bold"
              >
                            undefined undefined
              </span>
</Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <span>
                  00
</span>,
                <span
                  className="bold"
>
                  undefined undefined
</span>,
              ],
              "replace": false,
              "to": "/admin/about-me",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "00",
                },
                "ref": null,
                "rendered": "00",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "undefined undefined",
                  "className": "bold",
                },
                "ref": null,
                "rendered": "undefined undefined",
                "type": "span",
              },
            ],
            "type": [Function],
          },
          "type": "h2",
        },
        "type": "article",
      },
      Object {
        "instance": null,
        "key": "posts",
        "nodeType": "host",
        "props": Object {
          "children": <h2>
            <Link
                        replace={false}
                        to="/admin/posts"
            >
                        <span>
                                    00
                        </span>
                        <span
                                    className="bold"
                        >
                                    undefined undefined
                        </span>
            </Link>
</h2>,
          "className": "app-content-landing",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/admin/posts"
>
              <span>
                            00
              </span>
              <span
                            className="bold"
              >
                            undefined undefined
              </span>
</Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <span>
                  00
</span>,
                <span
                  className="bold"
>
                  undefined undefined
</span>,
              ],
              "replace": false,
              "to": "/admin/posts",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "00",
                },
                "ref": null,
                "rendered": "00",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "undefined undefined",
                  "className": "bold",
                },
                "ref": null,
                "rendered": "undefined undefined",
                "type": "span",
              },
            ],
            "type": [Function],
          },
          "type": "h2",
        },
        "type": "article",
      },
      Object {
        "instance": null,
        "key": "projects",
        "nodeType": "host",
        "props": Object {
          "children": <h2>
            <Link
                        replace={false}
                        to="/admin/projects"
            >
                        <span>
                                    00
                        </span>
                        <span
                                    className="bold"
                        >
                                    undefined undefined
                        </span>
            </Link>
</h2>,
          "className": "app-content-landing",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/admin/projects"
>
              <span>
                            00
              </span>
              <span
                            className="bold"
              >
                            undefined undefined
              </span>
</Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <span>
                  00
</span>,
                <span
                  className="bold"
>
                  undefined undefined
</span>,
              ],
              "replace": false,
              "to": "/admin/projects",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "00",
                },
                "ref": null,
                "rendered": "00",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "undefined undefined",
                  "className": "bold",
                },
                "ref": null,
                "rendered": "undefined undefined",
                "type": "span",
              },
            ],
            "type": [Function],
          },
          "type": "h2",
        },
        "type": "article",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <article
            className="app-content-landing"
>
            <h2>
                        <Link
                                    replace={false}
                                    to="/admin/about-me"
                        >
                                    <span>
                                                00
                                    </span>
                                    <span
                                                className="bold"
                                    >
                                                undefined undefined
                                    </span>
                        </Link>
            </h2>
</article>,
          <article
            className="app-content-landing"
>
            <h2>
                        <Link
                                    replace={false}
                                    to="/admin/posts"
                        >
                                    <span>
                                                00
                                    </span>
                                    <span
                                                className="bold"
                                    >
                                                undefined undefined
                                    </span>
                        </Link>
            </h2>
</article>,
          <article
            className="app-content-landing"
>
            <h2>
                        <Link
                                    replace={false}
                                    to="/admin/projects"
                        >
                                    <span>
                                                00
                                    </span>
                                    <span
                                                className="bold"
                                    >
                                                undefined undefined
                                    </span>
                        </Link>
            </h2>
</article>,
        ],
        "className": "app-content",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "about-me",
          "nodeType": "host",
          "props": Object {
            "children": <h2>
              <Link
                            replace={false}
                            to="/admin/about-me"
              >
                            <span>
                                          00
                            </span>
                            <span
                                          className="bold"
                            >
                                          undefined undefined
                            </span>
              </Link>
</h2>,
            "className": "app-content-landing",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/admin/about-me"
>
                <span>
                                00
                </span>
                <span
                                className="bold"
                >
                                undefined undefined
                </span>
</Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <span>
                    00
</span>,
                  <span
                    className="bold"
>
                    undefined undefined
</span>,
                ],
                "replace": false,
                "to": "/admin/about-me",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "00",
                  },
                  "ref": null,
                  "rendered": "00",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "undefined undefined",
                    "className": "bold",
                  },
                  "ref": null,
                  "rendered": "undefined undefined",
                  "type": "span",
                },
              ],
              "type": [Function],
            },
            "type": "h2",
          },
          "type": "article",
        },
        Object {
          "instance": null,
          "key": "posts",
          "nodeType": "host",
          "props": Object {
            "children": <h2>
              <Link
                            replace={false}
                            to="/admin/posts"
              >
                            <span>
                                          00
                            </span>
                            <span
                                          className="bold"
                            >
                                          undefined undefined
                            </span>
              </Link>
</h2>,
            "className": "app-content-landing",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/admin/posts"
>
                <span>
                                00
                </span>
                <span
                                className="bold"
                >
                                undefined undefined
                </span>
</Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <span>
                    00
</span>,
                  <span
                    className="bold"
>
                    undefined undefined
</span>,
                ],
                "replace": false,
                "to": "/admin/posts",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "00",
                  },
                  "ref": null,
                  "rendered": "00",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "undefined undefined",
                    "className": "bold",
                  },
                  "ref": null,
                  "rendered": "undefined undefined",
                  "type": "span",
                },
              ],
              "type": [Function],
            },
            "type": "h2",
          },
          "type": "article",
        },
        Object {
          "instance": null,
          "key": "projects",
          "nodeType": "host",
          "props": Object {
            "children": <h2>
              <Link
                            replace={false}
                            to="/admin/projects"
              >
                            <span>
                                          00
                            </span>
                            <span
                                          className="bold"
                            >
                                          undefined undefined
                            </span>
              </Link>
</h2>,
            "className": "app-content-landing",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/admin/projects"
>
                <span>
                                00
                </span>
                <span
                                className="bold"
                >
                                undefined undefined
                </span>
</Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <span>
                    00
</span>,
                  <span
                    className="bold"
>
                    undefined undefined
</span>,
                ],
                "replace": false,
                "to": "/admin/projects",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "00",
                  },
                  "ref": null,
                  "rendered": "00",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "undefined undefined",
                    "className": "bold",
                  },
                  "ref": null,
                  "rendered": "undefined undefined",
                  "type": "span",
                },
              ],
              "type": [Function],
            },
            "type": "h2",
          },
          "type": "article",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
